src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,496 @@
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.currentTextWriterContentType = contentType;
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            this.Flush();
+
+            this.textWriter = new ODataUtf8JsonTextWriter(this);
+
+            return this.textWriter;
+        }
+
+        /// <summary>
+        /// Ends a TextWriter value scope.
+        /// </summary>
+        public void EndTextWriterValueScope()
+        {
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+                this.textWriter.Dispose();
+                this.Flush();
+
+                CheckIfSeparatorNeeded();
+                CheckIfManualValueAtArrayStart();
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A task representing the asynchronous operation. The task result is a TextWriter for writing JSON data.</returns>
+        public async Task<TextWriter> StartTextWriterValueScopeAsync(string contentType)
+        {
+            await this.FlushAsync().ConfigureAwait(false);
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.currentTextWriterContentType = contentType;
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            await this.FlushAsync().ConfigureAwait(false);
+
+            this.textWriter = new ODataUtf8JsonTextWriter(this);
+
+            return this.textWriter;
+        }
+
+        /// <summary>
+        /// Asynchronously ends a TextWriter value scope.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task EndTextWriterValueScopeAsync()
+        {
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+                await this.textWriter.DisposeAsync();
+                await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+                CheckIfSeparatorNeeded();
+                CheckIfManualValueAtArrayStart();
+            }
+        }
+
+        /// <summary>
+        /// Whether the current TextWriter is writing JSON
+        /// </summary>
+        /// <returns></returns>
+        private bool IsWritingJson
+        {
+            get
+            {
+                return String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);

```

Comment:
Since this value is read multiple times during serialization, and since the value doesn't change, consider computing it only once and storing it in a field. I don't think there's any value in making it a property given that the property is also private.